src/api/java/com/minecolonies/api/colony/requestsystem/requestable/Food.java

```diff
@@ -102,18 +155,34 @@ public static void serialize(final IFactoryController controller, final PacketBu
      * @param buffer     the buffer to read.
      * @return the deliverable.
      */
-    public static Food deserialize(final IFactoryController controller, final PacketBuffer buffer)
-    {
+    public static Food deserialize(final IFactoryController controller, final PacketBuffer buffer) {
         final int count = buffer.readInt();
         final ItemStack result = buffer.readBoolean() ? buffer.readItemStack() : ItemStack.EMPTY;
 
+        List<ItemStorage> items = new ArrayList<>();
+        try
+        {
+            for (int i = 0; i < buffer.readInt(); ++i)
```

Comment:
now, this is the problem. You are reading the int from the buffer with each iteration. You need a size variable you declare beforehand to read the int too.

Reply:
vroniderps++